### ISSUE TITLE
the Small Object detection issue

### ISSUE THREAD

**Issue by xiaoxiongli** [sanghoon/pva-faster-rcnn#19]
Dear All, 

When i training my own dataset(229 category logo, totally 3w+ images) using PVANet, i got mAP = 0.9672, **indeed it is very high. PVANet is indeed so good**, the original R-FCN is 0.953(i do a modify it get to 0.9705). by the way, they use different backbone. so it is just a reference. but i find that **the small object** detection result is not as good as faster-rcnn and R-FCN. 

For **the Small object**,  the faster-rcnn get the best result, but its false alarm is also the highest. R-FCN get middle result and the lowest false alarm. PVANet get a worse small object detection result and middle false alarm result.

So, I feel confuse about this result. because of using the idea of HyperNet, the PVANet supposed to be have a good performance in small object detection, but it is not in my dataset...

Anyone please give me some tips about this issue, How can i fix it and get a better result?
@sanghoon 
@kyehyeon

**Comment by argman** [comment]
I think small objects maybe filtered by nms during generating proposals, you may check the nms parameter

**Comment by xiaoxiongli** [comment]
@argman Dear argman, i check the cfg file lib/fast_rcnn/config.py:

__C.TRAIN.RPN_MIN_SIZE = 16
__C.TEST.RPN_MIN_SIZE = 16

**it seems the parameter is OK**, So What's the problem about this issue?

**Comment by Simonhong111** [comment]
dear xiaoxiongli,I also want to train my own dataset with py-R-FCN,however i cannot modify my model correctly.Maybe, there are something else i  didn't know ,for example i didn't get the right pre-trained model ,could you please show me your models and give me  some advice.My Email address is 1475598891@qq.com 。thank you so much

**Comment by xiaoxiongli** [comment]
@Simonhong111 i am not use py-R-FCN, i implement the 4-stage training in MATLAB according the shaoqingren's faster-rcnn MATLAB code, I think you can ask the py-R-FCN code's author for help. i think after i done the PVANet, i will release the MATLAB version R-FCN 4-stage training code.

**Comment by argman** [comment]
@xiaoxiongli , the size is not controled by RPN_MIN_SIZE, the author implement a proposal_layer, you should set it in the prototxt like
layer {
  name: 'proposal'
  type: 'Proposal'
  bottom: 'rpn_cls_prob_reshape'
  bottom: 'rpn_bbox_pred'
  bottom: 'im_info'
  top: 'rpn_rois'
  top: 'rpn_scores'
  proposal_param {
    ratio: 0.5 ratio: 0.667 ratio: 1.0 ratio: 1.5 ratio: 2.0
    scale: 3 scale: 6 scale: 9 scale: 16 scale: 32
    base_size: 16
    feat_stride: 16
    pre_nms_topn: 12000
    post_nms_topn: 200
    nms_thresh: 0.7
    **min_size: 16**
  }
}

**Comment by sanghoon** [comment]
Hi @xiaoxiongli,
I guess maybe it's due to limited computational costs.
Could you share some quantitative comparison results?

**Comment by xiaoxiongli** [comment]
@sanghoon Dear sanghoon, sure, i already send you a e-mail with the test result and also some comment, thank you very much^_^...

**Comment by squidszyd** [comment]
Also curious about small object detection

**Comment by Caojunxu** [comment]
@xiaoxiongli Dear xiaoxiongli，have you solve the problem？

**Comment by xiaoxiongli** [comment]
@Caojunxu NO!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

**Comment by ck196** [comment]
@xiaoxiongli How small of objects do you need to detect? (16x16 is the smallest object?)
I also want to resolve this problem, SSD (single shot multibox detection) also has this problem.

**Comment by xiaoxiongli** [comment]
@ck196 above 24x24, or maybe above 32x32.................................... i send you a email and hope discuss with you when you are free

**Comment by AITTSMD** [comment]
@xiaoxiongli hi , I wonder how you finetune your own data? Use full/test.model or example_finetune384/? when I use test.model,I can't get a good result.When I use exampe_finetune384/,the speed is slow(0.18s/img,gtx1070).Can u give me some advice? Thank you in advance!

**Comment by xiaoxiongli** [comment]
@AITTSMD  example_finetune384, i use K40. in the aspect of speed, make sure that you do NOT compile caffe with cudnn, because cudnn's 1x1 convolution is slow.

**Comment by solemnrole** [comment]
hello,why the caffemodel th author provided i cant download it, do you success? if you hava download them, could you share with me?thanks a lot~

**Comment by MyVanitar** [comment]
@xiaoxiongli 

I appreciate if you have a look at my question: #63 
Also I appreciate if you inform me if you have published your MATLAB version R-FCN 4-stage training codes.

**Comment by shivam6294** [comment]
Facing the same `small object not detected` problem when using SSD. Was anyone able to find the cause? Did tweaking the nms parameter help? Would greatly appreciate it if someone could share their experience /solution attempts

**Comment by MyVanitar** [comment]
@shivam6294 

Increase the network image size to higher numbers

**Comment by FreeApe** [comment]
https://www.zhihu.com/question/49455386 ?
